File: tests/xcalablemp/global-view/task/C/barrier.c
#include <xmp.h>
#include <stdlib.h>
#include <stdio.h>   
#pragma xmp nodes p(*)

int main(void)
{
  if(xmp_num_nodes() <  4){
    printf("You have to run this program by more than 3 nodes.\n");
    exit(1);
  }

#pragma xmp task on p(1:3)
   {
#pragma xmp barrier
   }

#pragma xmp barrier
    
#pragma xmp task on p(1)
   {
     printf("PASS\n");
   }
   
   return 0;
}

File: tests/XMP/global-view/task/F/barrier.f90
program main
  include 'xmp_lib.h'
!$xmp nodes p(*)

  if(xmp_num_nodes().lt.4) then
     print *, 'You have to run this program by more than 3 nodes.'
     call exit(1)
  endif

!$xmp task on p(1:3)
!$xmp barrier
!$xmp end task

!$xmp barrier

!$xmp task on p(1)      
  write(*,*) "PASS"
!$xmp end task
end program main

